Adding a Sallen-Key lowpass filter to the Basic section.

slinkp opened the pull request:
Default settings:
- Freq range approx 100Hz-100kHz; midpoint approx 340Hz.
- Butterworth response (12dB/octave)

This required a dual-ganged pot, and per PR feedback I took the route of adding support for that to the netlist frontmatter handled by process-circuit.py.

This has a pair of extra resistors that are not in textbook SK implementations; these set an upper bound on the frequency range.

<img width="1113" height="813" alt="Screenshot 2026-04-29 at 11 55 39 AM" src="https://github.com/user-attachments/assets/4fb62bfd-8ac4-4ca1-90c6-2841b1f1829f" />

why-trv commented:
Awesome! Apologies for the late reply.

> Limitations: It would be much better with a dual-ganged pot. I have not been able to figure out whether/how that can be achieved with lcapy.

I'm not sure about lcapy either, but I think there has be a single slider in the UI anyway (e.g. the Blackstar HT5 model). So I guess its either editing the generated JS by hand or writing some sort of post-lcapy processing contraption. 

> It has a pair of extra resistors that are not in textbook SK implementations; these set an upper bound on the frequency range. Maybe should remove them and leave it standard?

I'm guessing real-world hardware has these extra resistors? If so, I think we should leave them in.

---

We'll also need to tweak the lcapy netlist to make it draw nicely without those right angles near opamp inputs.

slinkp commented:
Thanks for replying on this and sorry for not addressing the feedback - I will eventually :D ... just suffering from self-induced project overload lately.

slinkp committed:
Adding a Sallen-Key lowpass filter to the Basic section.

Default settings:
- Freq range approx 100Hz-50kHz
- Butterworth response (12dB/octave)

It has a pair of resistors that are not on textbook SK implementations;
these set an upper bound on the frequency range.

Required adding support for dual-ganged controls in process-circuit.py
and documenting that in circuits/README.md.

slinkp commented:
@why-trv I finally updated this based on your feedback.  PR description updated with new screenshot.  

Re "tweak the lcapy netlist to make it draw nicely without those right angles near opamp inputs" ... controlling lcapy drawing seems vastly under-documented and very hard; I wanted to get the op amp smaller, but the best I could figure out was this exact magic combination of `size` and `scale`. Whatever tricks you used on BossFZ2EQ.sch I was not able to figure out (nor was Cursor).

Also, I wanted to remove `E, E_Ac`, and `E_Ro` as user-controlled parameters, but that seems to be impossible given that I had to provide E_Ac and E_Ro to avoid crashes. 🤷
